mx.jruby/mx_jruby.py

```diff
@@ -0,0 +1,113 @@
+import mx
+import os
+from os.path import join, exists
+import subprocess
+import shutil
+
+_suite = mx.suite('jruby')
+
+class MavenProject(mx.Project):
+    def __init__(self, suite, name, deps, workingSets, theLicense, **args):
+        mx.Project.__init__(self, suite, name, "", [], deps, workingSets, _suite.dir, theLicense)
+        self.javaCompliance = "1.7"
+        self.build = hasattr(args, 'build')
+        self.prefix = args['prefix']
+
+    def source_dirs(self):
+        return []
+
+    def output_dir(self):
+        dir = os.path.join(_suite.dir, self.prefix)
+        return dir[0:len(dir) - 1]
+
+    def source_gen_dir(self):
+        return None
+
+    def getOutput(self, replaceVar=False):
+        return os.path.join(_suite.dir, "target")
+
+    def getResults(self, replaceVar=False):
+        return mx.Project.getResults(self, replaceVar=replaceVar)
+
+    def getBuildTask(self, args):
+        return MavenBuildTask(self, args, None, None)
+
+    def isJavaProject(self):
+        return True
+
+    def archive_prefix(self):
+        return self.prefix
+
+
+class MavenBuildTask(mx.BuildTask):
+    def __init__(self, project, args, vmbuild, vm):
+        mx.BuildTask.__init__(self, project, args, 1)
+        self.vm = vm
+        self.vmbuild = vmbuild
+
+    def __str__(self):
+        return 'Building Maven for {}'.format(self.subject)
+
+    def needsBuild(self, newestInput):
+        return (True, 'Let us re-build everytime')
+
+    def newestOutput(self):
+        return None
+
+    def build(self):
+        if not self.subject.build:
+            mx.log("...skip build of {}".format(self.subject))
+            return
+        mx.log('...perform build of {}'.format(self.subject))
+
+        rubyDir = _suite.dir
+
+        # HACK: since the maven executable plugin does not configure the
+        # java executable that is used we unfortunately need to append it to the PATH
+        javaHome = os.getenv('JAVA_HOME')
+        if javaHome:
+            os.environ["PATH"] = os.environ["JAVA_HOME"] + '/bin' + os.pathsep + os.environ["PATH"]
+
+        mx.logv('Setting PATH to {}'.format(os.environ["PATH"]))
+        mx.logv('Calling java -version')
+        mx.run(['java', '-version'])
+
+        # Truffle version
+
+        truffle = mx.suite('truffle')
+        truffle_commit = truffle.vc.parent(truffle.dir)
+
+        mx.run_mx(['maven-install'], suite=truffle)
+
+        # Ruby version
+        def apply_to_file(filename, function):
+            contents = open(filename).read()
+            contents = function(contents)
+            open(filename, 'w').write(contents)
+        apply_to_file(join(rubyDir, 'VERSION'), lambda version: 'graal-vm\n')
```

Comment:
In this case, maybe a simple `open(filename, 'w').write("graalvm\n")` is enough?